### PR TITLE
Use sts tcptracer bpf

### PR DESCRIPTION
**Pull request by zandrewitte**
Includes changes that allows us to give directionality to connections that the ebpf-tracer discovers.

### CODE REVIEW

File: conf-dev.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ process_config:
 #  process_dd_url: http://localhost:7077/stsAgent
 
 # Use this when running process agent in the vagrant vms
-  process_sts_url: http://192.168.56.1:7077/stsAgent
+  process_dd_url: http://192.168.56.1:7077/stsAgent
```
Comment (TizianoPerrucci) [inline conf-dev.yaml]: why are we renaming this back to `_dd_`, should not be the case

File: glide.yaml [hunk 2]
```diff
@@ -2,9 +2,9 @@ package: github.com/StackVista/stackstate-process-agent
 import:
   - package: github.com/DataDog/datadog-agent
     version: f7e28d260d8f491189c4ddf377c36c7ccb55e89c
-  - package: github.com/DataDog/tcptracer-bpf
+  - package: github.com/StackVista/tcptracer-bpf
     vcs: git
-    version: dd
+    version: add-connection-direction
```
Comment (TizianoPerrucci) [inline glide.yaml]: should this eventually be `master` ?

Comment (zandrewitte) [reply glide.yaml]: Yes, we should update it to master after merging the `add-connection-direction` branch

File: conf-dev.yaml (hunk 1 above)
Comment (zandrewitte) [reply conf-dev.yaml]: The `apply_branding.sh` script will change it to `process_sts_url` during the build process

Comment (TizianoPerrucci) [reply conf-dev.yaml]: cool

File: glide.yaml (hunk 2 above)
Comment (TizianoPerrucci) [reply glide.yaml]: ok

File: checks/net.go [hunk 3]
```diff
@@ -169,6 +169,7 @@ func (c *ConnectionsCheck) formatConnections(conns []tracer.ConnectionStats, las
 			},
 			BytesSent:     calculateRate(conn.SendBytes, lastConns[key].SendBytes, lastCheckTime),
 			BytesRecieved: calculateRate(conn.RecvBytes, lastConns[key].RecvBytes, lastCheckTime),
+			Incoming:      isIncoming(conn.Direction),
```
Comment (craffit) [inline checks/net.go]: This should become a tri-state. (Incoming/ outgoing/ unknown). These are the 3 states that come out of tcptracer bpf and stackstate deals with

File: checks/net.go [hunk 4]
```diff
@@ -197,6 +198,15 @@ func formatType(f tracer.ConnectionType) model.ConnectionType {
 	}
 }
 
+func isIncoming(d tracer.Direction) bool {
+	switch d {
+	case tracer.INCOMING:
+		return true
+	default:
+		return false
```
Comment (craffit) [inline checks/net.go]: Both unknown and outgoing direction map to false. This is not good i think

File: proto/agent.proto [hunk 5]
```diff
@@ -321,8 +321,8 @@ message Connection {
 	ConnectionFamily family = 10;
 	ConnectionType type = 11;
 	int64 pidCreateTime = 12;
-	bool incoming = 51;
 	string namespace = 50; // local
+	bool incoming = 51;
```
Comment (craffit) [inline proto/agent.proto]: I think we should turn this into an enum ( or use absense of the field as unknown)

**Pull request merged**